src/test/java/app/sami/languageWeb/language/LanguageControllerTests.java

```diff
@@ -0,0 +1,189 @@
+        languageGradesRepository.save(LanguageGradesFactory.generateFromUsers(
+                userTest2.getId(), userTest1.getId(), grade1, Language.ROMANIAN));
+        languageGradesRepository.save(LanguageGradesFactory.generateFromUsers(
+                userTest2.getId(), userTest2.getId(), grade1, Language.MACEDONIAN));
+        languageGradesRepository.save(LanguageGradesFactory.generateFromUsers(
+                userTest2.getId(), userTest2.getId(), grade1, Language.FAROESE));
+    }
+
+    @Test
+    void getUserGradeStats_ReturnsUserStatsAnd200() throws Exception {
+
+        Language language = Language.ARABIC;
+        UUID userId = userTest1.getId();
+        String url = "/public/users/" + userId +  "/" + language + "/user-grade";
+        mockMvc.perform(get(url))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.avgGrade").value(75));
+    }
+
+    @Test
+    void getUserGradeStatsByEvaluator_ReturnsUserStatsAnd200() throws Exception {
+        UUID userId = userTest1.getId();
+        Language language = Language.ENGLISH;
+        String url = "/public/users/" + userId + "/" + language + "/evaluator-grade";
+        mockMvc.perform(get(url))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.avgGrade").value(75))
+                .andExpect(jsonPath("$.gradeCount").value(2));
+    }
+
+    @Test
+    void givenNoUserGradeStatsByEvaluator_Returns404() throws Exception {
+        String url = "/public/users/" + UUID.randomUUID() + "/" + Language.ENGLISH + "/evaluator-grade";
+        mockMvc.perform(get(url))
+                .andExpect(status().isNotFound());
+
+    }
+    @Test
+    void givenWrongRouteReturns4xx()throws Exception{
+        Language language = Language.ARABIC;
+        UUID userId = UUID.randomUUID();
+        String url = "/public/users/" + userId +  "/" + language + "/user-grades";
+        mockMvc.perform(get(url))
+                .andExpect(status().is4xxClientError());
+    }
+
+    @Test
+    void getUserLanguages_ReturnsLanguageGradesListAnd200() throws Exception{
+        UUID userId = userTest2.getId();
+        String url = "/public/users/" + userId + "/languages";
+        mockMvc.perform(get(url))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void createLanguageGrade_ReturnsLanguageGradeAnd200() throws Exception {
+        UUID userId = userTest2.getId();
+        UUID emitterId = userTest1.getId();
+        String token = authUser(userTest1);
+        String url = "/users/create-rating";
+        LanguageGradeRequest languageGradeRequest = LanguageGradeRequest
+                .builder()
+                        .grade(grade1)
+                                .emitterUserId(emitterId)
+                                        .language(Language.MACEDONIAN)
+                .userId(userId)
+                                                .build();
+        mockMvc.perform(post(url)
```

Comment:
you can make more protected method to avoid repeating the objectMapper and auth thing:

```
post(String url, Object content);
post(String url, String token, Object content);
```